Tests/Common/Data/VolumeRenkoConsolidatorTests.cs

```diff
@@ -0,0 +1,240 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+using Python.Runtime;
+using QuantConnect.Data.Consolidators;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Tests.Common.Data
+{
+    [TestFixture]
+    public class VolumeRenkoConsolidatorTests
+    {
+        [Test]
+        public void OutputTypeIsVolumeRenkoBar()
+        {
+            var consolidator = new VolumeRenkoConsolidator(10);
+            Assert.AreEqual(typeof(VolumeRenkoBar), consolidator.OutputType);
+        }
+
+        [Test]
+        public void ConsolidatesOnTickVolumeReached()
+        {
+            VolumeRenkoBar bar = null;
+            var consolidator = new VolumeRenkoConsolidator(10);
+            consolidator.DataConsolidated += (sender, consolidated) =>
+            {
+                bar = consolidated;
+            };
+
+            var reference = DateTime.Today;
+            consolidator.Update(new Tick(reference, Symbol.Empty, String.Empty, String.Empty, 2m, 1m));
+            Assert.IsNull(bar);
+
+            consolidator.Update(new Tick(reference.AddHours(1), Symbol.Empty, String.Empty, String.Empty, 3m, 2m));
+            Assert.IsNull(bar);
+
+            consolidator.Update(new Tick(reference.AddHours(2), Symbol.Empty, String.Empty, String.Empty, 3m, 3m));
+            Assert.IsNull(bar);
+
+            consolidator.Update(new Tick(reference.AddHours(3), Symbol.Empty, String.Empty, String.Empty, 2m, 2m));
+            Assert.IsNotNull(bar);
+
+            Assert.AreEqual(1m, bar.Open);
+            Assert.AreEqual(3m, bar.High);
+            Assert.AreEqual(1m, bar.Low);
+            Assert.AreEqual(2m, bar.Close);
+            Assert.AreEqual(reference.AddHours(3), bar.EndTime);
+            Assert.IsTrue(bar.IsClosed);
```

Comment:
To be safe should assert all of the properties startTime/symbol/etc/the volume! same for the other tests 👍